src/api/services/userprofile.py

```diff
@@ -152,8 +153,26 @@ def add_action_completed(self, context, args) -> (dict, MassEnergizeAPIError):
       return None, err
     return serialize(user, full=True), None
   
-  def handle_csv(self, context, args, temporarylocation) -> (dict, MassEnergizeAPIError):
-    info, err = self.store.handle_csv(context, args, temporarylocation)
-    if err:
-      return None, err
-    return info, None
\ No newline at end of file
+  def handle_csv(self, context, args, filecontents) -> (dict, MassEnergizeAPIError):
+    first_name_field = args.get('first_name_field', None)
+    email_field = args.get('email_field', None)
+    invalid_emails = []
+    for csv_row in filecontents:
+      column_list = list(csv_row.keys())
+      try:
+        # prevents the first row (headers) from being read in as a user
+        if csv_row[first_name_field] == column_list[0]:
+          continue
+        # verify correctness of email address
+        regex = '^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$'
```

Comment:
again a helper function called `is_valid_email` would be great so that other functions can call that same function in other places.